File: 383-ransom-note/383-ransom-note.c

bool canConstruct(char* ransomNote, char* magazine) {

    int r_len=strlen(ransomNote);
    int m_len=strlen(magazine);
    
    int alphabet[26]={0};
    
    for(int i=0; i<r_len; i++){
        
        alphabet[ransomNote[i]-'a']--;
        
    }
    
    for(int i=0; i<m_len; i++){
        
        alphabet[magazine[i]-'a']++;
        
    }
    
    for(int i=0; i<26; i++){
        
        if(alphabet[i]<0)
            return false;
    }
    
    return true;
}

File: 383-ransom-note/383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        
        a = Counter(ransomNote)
        b = Counter(magazine)
        
        for i, _ in a.items():

            if(a[i]<=b[i]):
                continue
            else:
                return False
        return True